### DISCUSSION TITLE
Usage advice for in browser code editor phcode.dev/brackets.io

### DISCUSSION THREAD

**Discussion by charlypa** [copy/v86#796] [Q&A]
We are currently searching for an easy-to-boot Linux box with `node`, `gcc`  and `python`  for a web-based code editor [phcode.dev](https://phcode.dev). Students use this code editor to learn to program. I have recently evaluated sample  Linux images.
It looks promising.
Would you help us to answer the following questions
1. Is this stable enough to adopt?
2. What is the right Linux image for better performance for `node` `python` and `gcc` ?

**Comment by abose** [comment]
https://github.com/phcode-dev/phoenix is the codebase. The main aim is to provide easy-to-use virtual environments for students to test and learn various available languages and compilers. 
Also, what would be the advice on extracting maximum performance out of the system, which image would be the most performant?

**Comment by cemalgnlts** [comment]
Hi,

You can try ArchLinux (https://copy.sh/v86/?profile=archlinux) but performance may be problematic:

`gcc`
![Screenshot 2023-05-12 18 25 15](https://github.com/copy/v86/assets/45357531/717bfb8b-1954-4634-83d4-729c591b6cec)

`python3`
![Screenshot 2023-05-12 18 26 57](https://github.com/copy/v86/assets/45357531/3ce71634-93e8-4422-89e1-6e53d63130a0)

`node` takes about 2 minutes. I am also making arrangements to run node in the browser, if you leave node in repl mode it will run faster but it still takes ~10-20 seconds.

You can also try different methods:

`Pyodide`: https://pyodide.org/en/stable/
`Nodebox`: https://github.com/codesandbox/nodebox-runtime
`WebContainers`: https://webcontainers.io/

**Reply by cemalgnlts** [reply]
Tiny Core Linux `node` is better. I didn't test it, but it's worth a try for `python3` and `gcc`.
